### PR TITLE
Add eslint

### PR DESCRIPTION
**Pull request by trentmwillis**
Introduces a simple ESLint setup to help us catch silly mistakes.

### CODE REVIEW

File: scripts/precook-node-modules.js [hunk 1]
```diff
@@ -23,5 +23,5 @@ fs.ensureDir('tmp')
     symlinkDirectory(root, path.join(precooked, 'ember-cli-fastboot'));
   })
   .catch(function(e) {
-    console.log(e);
+    console.log(e); // eslint-disable-line no-console
```
Comment (rwjblue) [inline scripts/precook-node-modules.js]: For future PR: should this rethrow?

Comment (trentmwillis) [inline scripts/precook-node-modules.js]: Unsure, I'm not even entirely sure what this module is for (I don't see it being used anywhere).

Comment (simonihmig) [inline scripts/precook-node-modules.js]: https://github.com/ember-fastboot/ember-cli-fastboot/blob/master/package.json#L19 😬

**Comment by simonihmig** [comment]
This should also run in CI. Currently `npm test` is configured to only run `test/acceptance`: https://github.com/tomdale/ember-cli-addon-tests/blob/master/package.json#L20

File: scripts/precook-node-modules.js (hunk 1 above)
Comment (trentmwillis) [inline scripts/precook-node-modules.js]: Seems like it should live in ember-cli-fastboot then...anyway, I'll leave it be for now

**Comment by trentmwillis** [comment]
Good point, I had updated that, but it was on another branch.

**Comment by trentmwillis** [comment]
Updated so the lint tests actually run now

File: scripts/precook-node-modules.js (hunk 1 above)
Comment (simonihmig) [inline scripts/precook-node-modules.js]: Agreed. Basically everybody starting to work on this repo some time stumbles upon this. Same for me 😆

**Pull request merged**